src/runtime/d3d12compute.cpp

```diff
@@ -2590,7 +2594,7 @@ using namespace Halide::Runtime::Internal::D3D12Compute;
 WEAK bool enable_allocation_cache = false;
 static const int MaxBuffersInCache = 32;
 WEAK d3d12_buffer *buffer_pool[MaxBuffersInCache] = {};
-WEAK halide_mutex buffer_pool_lock;
+static halide_mutex buffer_pool_lock;
```

Comment:
Why this change? It will result in multiple locks existing when people merge runtimes.

Reply:
Does weak linkage even work in clang-cl (which is necessary to compile the d3d12 compute modules)? By merging runtimes, you mean merging debug and non-debug together?

Reply:
The lock should certainly have the same linkage class as the things the lock is protecting. Weak linkage gets converted to something equivalent to inline linkage, which is handled on windows via a comdat.

By merging runtimes I mean if you compile without -no_runtime and then link together multiple Halide pipelines.

Reply:
I see, yeah, makes sense then.
I'll probably have to revisit the global variables and symbols in the d3d12 runtime to make sure I did not miss any weak decoration.